app/dao/notifications_dao.py

```diff
@@ -317,21 +319,43 @@ def delete_notifications_created_more_than_a_week_ago_by_type(notification_type)
     deleted = 0
     for f in flexible_data_retention:
         days_of_retention = convert_utc_to_bst(datetime.utcnow()).date() - timedelta(days=f.days_of_retention)
-        deleted += db.session.query(Notification).filter(
+        query = db.session.query(Notification).filter(
             func.date(Notification.created_at) < days_of_retention,
-            Notification.notification_type == f.notification_type,
-            Notification.service_id == f.service_id
-        ).delete(synchronize_session='fetch')
+            Notification.notification_type == f.notification_type, Notification.service_id == f.service_id)
+        _delete_letters_from_s3(notification_type, query)
+        deleted += query.delete(synchronize_session='fetch')
+
     seven_days_ago = convert_utc_to_bst(datetime.utcnow()).date() - timedelta(days=7)
     services_with_data_retention = [x.service_id for x in flexible_data_retention]
-    deleted = db.session.query(Notification).filter(
-        func.date(Notification.created_at) < seven_days_ago,
-        Notification.notification_type == notification_type,
-        Notification.service_id.notin_(services_with_data_retention)
-    ).delete(synchronize_session='fetch')
+    query = db.session.query(Notification).filter(func.date(Notification.created_at) < seven_days_ago,
+                                                  Notification.notification_type == notification_type,
+                                                  Notification.service_id.notin_(
+                                                      services_with_data_retention))
+    _delete_letters_from_s3(notification_type=notification_type, query=query)
+    deleted = query.delete(synchronize_session='fetch')
     return deleted
 
 
+def _delete_letters_from_s3(notification_type, query):
+    if notification_type == LETTER_TYPE:
```

Comment:
This is minor, but I'd probably move this condition to the task function. Then you won't need the `notification_type` argument and it's clear right away that this doesn't do anything for other notification types.